src/include/duckdb/main/attached_database.hpp

```diff
@@ -104,6 +120,10 @@ class AttachedDatabase : public CatalogEntry {
 	optional_ptr<StorageExtension> storage_extension;
 	bool is_initial_database = false;
 	bool is_closed = false;
+
+	//! key ID for encryption
+	string encryption_key_id;
```

Comment:
This should probably be in the `DuckCatalog` - `AttachedDatabase` is a generic class that is the back-end of many different databases.